### ISSUE TITLE
Comments are showing up twice

### ISSUE THREAD

**Issue by amj311** [microsoft/vscode-pull-request-github#3978]
Type: <b>Bug</b>

After the recent VSCode update that changed the styles of the in-editor PR comments UI, the comments from my PRs having been showing up with duplicates, both in the editor itself and in the "Comments" drawer. In the comments drawer filter input, the total says "Showing 6 of 4", which definitely indicates some error.

I do not know if this is because of interference with another extension or anything like that - I also have GitLens installed.

As far as I can tell, these are the steps to reproduce:
- Checkout a branch with an open PR and comments
- Open the "Comments" drawer and view the PR comments
  - The comments will be duplicated.
  - <img width="1438" alt="Screen Shot 2022-09-22 at 4 23 28 PM" src="https://user-images.githubusercontent.com/38047053/191861831-f3a1351c-a7a9-45c3-b2dc-cb71a5f064bd.png">

- Click on any of the comments to open it in the editor
  - The comment will appear twice at the commented line.
  - <img width="872" alt="Screen Shot 2022-09-22 at 4 24 56 PM" src="https://user-images.githubusercontent.com/38047053/191861987-1b56195d-a060-4b78-ad4e-4ae0d930ecac.png">

Extension version: 0.50.0
VS Code version: Code 1.71.2 (Universal) (74b1f979648cc44d385a2286793c226e611f59e7, 2022-09-14T21:05:37.721Z)
OS version: Darwin x64 20.6.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|4, 3, 3|
|Memory (System)|16.00GB (0.61GB free)|
|Process Argv|--crash-reporter-id 5a87eb07-c55f-4254-9796-cd8121b0b85e|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30557514
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
hb751961:30553087
dsvsc012:30540252
azure-dev_surveyonecf:30548226
i497e931:30553904
fc301958:30567733

```

</details>

**Comment by Thomas1664** [comment]
I had the same issue a few days ago but I can't reproduce it.

**Comment by amj311** [comment]
@alexr00 Unfortunately it was a private repository and can't be shared. I also don't use the feature very often so I haven't seen it since, but it was consistent for that entire week or so.

This may be helpful: I believe their were comments from multiple pull requests on the same branch - a previously closed one and a new one. Perhaps that had something to do with it?

Sorry I don't have much more info for you!

**Comment by alexr00** [comment]
I tried with multiple PRs from the same branch, but I still can't reproduce this issue. Leaving open for now to see if others have ideas on how to trigger this issue.

**Comment by alexr00** [comment]
I found a case where this happens and fix it with [21db0b3](https://github.com/microsoft/vscode-pull-request-github/commit/21db0b37c9e268dfcf7a508a3af91058fe3e5124)
